src/mongoc/mongoc-client.c

```diff
@@ -672,9 +662,48 @@ _mongoc_client_recv_gle (mongoc_client_t  *client,
  *
  *--------------------------------------------------------------------------
  */
+mongoc_client_t *
+mongoc_client_new(const char *uri_string)
+{
+   mongoc_uri_t *uri;
+   mongoc_topology_t *topology;
+
+   if (!uri_string) {
+      uri_string = "mongodb://127.0.0.1/";
+   }
+
+   if (!(uri = mongoc_uri_new(uri_string))) {
+      return NULL;
+   }
+
+   topology = mongoc_topology_new(uri);
+
+   mongoc_uri_destroy (uri);
+
+   return _mongoc_client_new(uri_string, topology);
+}
```

Comment:
It's because of the way the topology object must be handled.  Client creation either happens by users calling ```mongoc_client_new()``` directly, or happens through a client pool, which calls the internal ```_mongoc_client_new()```.  When clients are created through a pool, they share a single topology object that the pool owns.  When clients are created directly, they each create and own separate topology objects.  This wrapper handles the creation of the topology object and then calls the internal method to complete client setup.